### PR TITLE
Cinnamon 4.8

### PR DESCRIPTION
**Pull request by SparkyBluefang**
https://bugs.gentoo.org/757708

This updates the Cinnamon collection of packages to the latest released 4.8.x versions.

* A few packages were converted to meson.
* I went through a attempted to clean-up the dependencies. They should be alphabetic and I attempted to remove unused and add missing runtime deps.
* There are a couple patches, all of which have been merged (unreleased) or have open PRs. The cjs patch is a little weird because repoman was complaining it was too large, so it's downloading the patch from GitHub.

### CODE REVIEW

**Comment by SparkyBluefang** [comment]
Also worth noting that I added python 3_9 targets where possible, though have not yet thoroughly validated it yet.

Updating nemo was blocked by `dev-python/polib` missing the 3_9 target. Is that something that can be done as a drive-by update here? Or should a bug be filed against the package?

File: gnome-extra/cinnamon-control-center/cinnamon-control-center-4.8.2.ebuild
Comment (juippis) [inline gnome-extra/cinnamon-control-center/cinnamon-control-center-4.8.2.ebuild:8]: Please add the Github page as 2ndary HOMEPAGE here.

File: gnome-extra/cinnamon-control-center/cinnamon-control-center-4.8.2.ebuild
Comment (juippis) [inline gnome-extra/cinnamon-control-center/cinnamon-control-center-4.8.2.ebuild:5]: Empty line below EAPI.

File: gnome-extra/cinnamon-desktop/cinnamon-desktop-4.8.1.ebuild
Comment (juippis) [inline gnome-extra/cinnamon-desktop/cinnamon-desktop-4.8.1.ebuild:33]: Looks like `COMMON_DEPEND` is useless here. Just define it as `RDEPEND` and use it in `DEPEND`.

File: gnome-extra/cinnamon-control-center/cinnamon-control-center-4.8.2.ebuild
Comment (juippis) [inline gnome-extra/cinnamon-control-center/cinnamon-control-center-4.8.2.ebuild:30]: No empty line please.

File: gnome-extra/cinnamon-control-center/cinnamon-control-center-4.8.2.ebuild
Comment (juippis) [inline gnome-extra/cinnamon-control-center/cinnamon-control-center-4.8.2.ebuild:39]: Here too, no empty line.

File: gnome-extra/cinnamon-menus/cinnamon-menus-4.8.2.ebuild
Comment (juippis) [inline gnome-extra/cinnamon-menus/cinnamon-menus-4.8.2.ebuild:28]: No newline here please.

File: gnome-extra/cinnamon-desktop/cinnamon-desktop-4.8.1.ebuild
Comment (juippis) [inline gnome-extra/cinnamon-desktop/cinnamon-desktop-4.8.1.ebuild:11]: Github page as 2ndary HOMEPAGE please.

File: gnome-extra/cinnamon-menus/cinnamon-menus-4.8.2.ebuild
Comment (juippis) [inline gnome-extra/cinnamon-menus/cinnamon-menus-4.8.2.ebuild:9]: Github page as 2ndary HOMEPAGE.

File: gnome-extra/cinnamon-screensaver/cinnamon-screensaver-4.8.1.ebuild
Comment (juippis) [inline gnome-extra/cinnamon-screensaver/cinnamon-screensaver-4.8.1.ebuild:11]: Add Github page too.

File: gnome-extra/cinnamon-screensaver/cinnamon-screensaver-4.8.1.ebuild
Comment (juippis) [inline gnome-extra/cinnamon-screensaver/cinnamon-screensaver-4.8.1.ebuild:32]: No newline needed.

File: gnome-extra/cinnamon-screensaver/cinnamon-screensaver-4.8.1.ebuild
Comment (juippis) [inline gnome-extra/cinnamon-screensaver/cinnamon-screensaver-4.8.1.ebuild:43]: Here too.

File: gnome-extra/cinnamon-session/cinnamon-session-4.8.0.ebuild
Comment (juippis) [inline gnome-extra/cinnamon-session/cinnamon-session-4.8.0.ebuild:9]: Github page as 2ndary HOMEPAGE.

File: gnome-extra/cinnamon-session/cinnamon-session-4.8.0.ebuild
Comment (juippis) [inline gnome-extra/cinnamon-session/cinnamon-session-4.8.0.ebuild:34]: No emptyline needed.

File: gnome-extra/cinnamon-session/cinnamon-session-4.8.0.ebuild
Comment (juippis) [inline gnome-extra/cinnamon-session/cinnamon-session-4.8.0.ebuild:42]: Looks like `COMMON_DEPEND` is not needed here. You can first define `DEPEND`, then use that as `RDEPEND`.

File: gnome-extra/cinnamon-session/cinnamon-session-4.8.0.ebuild
Comment (juippis) [inline gnome-extra/cinnamon-session/cinnamon-session-4.8.0.ebuild:46]: No newline here.

File: gnome-extra/cinnamon-settings-daemon/cinnamon-settings-daemon-4.8.3.ebuild
Comment (juippis) [inline gnome-extra/cinnamon-settings-daemon/cinnamon-settings-daemon-4.8.3.ebuild:8]: Add Github page.

File: gnome-extra/cinnamon-settings-daemon/cinnamon-settings-daemon-4.8.3.ebuild
Comment (juippis) [inline gnome-extra/cinnamon-settings-daemon/cinnamon-settings-daemon-4.8.3.ebuild:37]: No empty line needed.

File: gnome-extra/cinnamon-translations/cinnamon-translations-4.8.3.ebuild
Comment (juippis) [inline gnome-extra/cinnamon-translations/cinnamon-translations-4.8.3.ebuild:8]: These days we install all localization files unconditionally, but up to you whether you want to update it here or not. But you'll be required to update this at some point. 

https://projects.gentoo.org/qa/policy-guide/installed-files.html#pg0301

File: gnome-extra/cinnamon-translations/cinnamon-translations-4.8.3.ebuild
Comment (juippis) [inline gnome-extra/cinnamon-translations/cinnamon-translations-4.8.3.ebuild:11]: Add Github page.

File: gnome-extra/cinnamon/cinnamon-4.8.5.ebuild
Comment (juippis) [inline gnome-extra/cinnamon/cinnamon-4.8.5.ebuild:11]: Add Github page.

File: gnome-extra/cinnamon/cinnamon-4.8.5.ebuild
Comment (juippis) [inline gnome-extra/cinnamon/cinnamon-4.8.5.ebuild:40]: No empty line needed.

File: gnome-extra/cinnamon/cinnamon-4.8.5.ebuild
Comment (juippis) [inline gnome-extra/cinnamon/cinnamon-4.8.5.ebuild:5]: Empty line below EAPI.

File: gnome-extra/cinnamon/cinnamon-4.8.5.ebuild
Comment (juippis) [inline gnome-extra/cinnamon/cinnamon-4.8.5.ebuild:96]: These are all `BDEPEND`, makes it looks like you don't need `COMMON_DEPEND` here.

File: gnome-extra/cinnamon/cinnamon-4.8.5.ebuild
Comment (juippis) [inline gnome-extra/cinnamon/cinnamon-4.8.5.ebuild:86]: No empty line here.

File: gnome-extra/cinnamon/cinnamon-4.8.5.ebuild
Comment (juippis) [inline gnome-extra/cinnamon/cinnamon-4.8.5.ebuild:95]: No empty line here.

File: gnome-extra/cinnamon-settings-daemon/cinnamon-settings-daemon-4.8.3.ebuild
Comment (juippis) [inline gnome-extra/cinnamon-settings-daemon/cinnamon-settings-daemon-4.8.3.ebuild:5]: Empty line below EAPI please.

File: gnome-extra/cinnamon/cinnamon-4.8.5.ebuild
Comment (juippis) [inline gnome-extra/cinnamon/cinnamon-4.8.5.ebuild:122]:
```suggestion
	for p in $(grep -rl '#!.*python3' || die); do
```

File: gnome-extra/cinnamon/cinnamon-4.8.5.ebuild
Comment (juippis) [inline gnome-extra/cinnamon/cinnamon-4.8.5.ebuild:145]: We avoid using `rm -f` because it hides errorneous calls and then they're just left in the ebuild for an eternity. So remove the `-f`, it's only used if something can't be predicted.

File: gnome-extra/cjs/cjs-4.8.1.ebuild
Comment (juippis) [inline gnome-extra/cjs/cjs-4.8.1.ebuild:5]: Empty line below EAPI.

File: gnome-extra/cjs/cjs-4.8.1.ebuild
Comment (juippis) [inline gnome-extra/cjs/cjs-4.8.1.ebuild:8]: Add Github page.

File: gnome-extra/cjs/cjs-4.8.1.ebuild
Comment (juippis) [inline gnome-extra/cjs/cjs-4.8.1.ebuild:24]: No empty line here.

File: gnome-extra/cjs/cjs-4.8.1.ebuild
Comment (juippis) [inline gnome-extra/cjs/cjs-4.8.1.ebuild:32]: No empty line here.

File: gnome-extra/nemo/nemo-4.8.4.ebuild
Comment (juippis) [inline gnome-extra/nemo/nemo-4.8.4.ebuild:35]: No empty line here.

File: gnome-extra/nemo/nemo-4.8.4.ebuild
Comment (juippis) [inline gnome-extra/nemo/nemo-4.8.4.ebuild:46]: No empty line here needed.

File: gnome-extra/nemo/nemo-4.8.4.ebuild
Comment (juippis) [inline gnome-extra/nemo/nemo-4.8.4.ebuild:11]: Add Github page.

File: x11-libs/xapps/xapps-2.0.5.ebuild
Comment (juippis) [inline x11-libs/xapps/xapps-2.0.5.ebuild:38]: I think these are all `BDEPEND`.

File: x11-libs/xapps/xapps-2.0.5.ebuild
Comment (juippis) [inline x11-libs/xapps/xapps-2.0.5.ebuild:71]: Drop the `-f`.

File: x11-libs/xapps/xapps-2.0.5.ebuild
Comment (juippis) [inline x11-libs/xapps/xapps-2.0.5.ebuild:81]: Add `|| die`.

File: x11-libs/xapps/xapps-2.0.5.ebuild
Comment (juippis) [inline x11-libs/xapps/xapps-2.0.5.ebuild:79]: You can `|| die` inside the subshell `$( )`. `${EPYTHON}` is probably what you're looking for.

File: x11-libs/xapps/xapps-2.0.5.ebuild
Comment (juippis) [inline x11-libs/xapps/xapps-2.0.5.ebuild:80]:
```suggestion
		einfo "gobject overrides directory: ${PYTHON_GI_OVERRIDESDIR}"
```

File: x11-libs/xapps/xapps-2.0.5.ebuild
Comment (juippis) [inline x11-libs/xapps/xapps-2.0.5.ebuild:81]:
```suggestion
		mkdir -p "${ED}/${PYTHON_GI_OVERRIDESDIR}/"
```

File: x11-libs/xapps/xapps-2.0.5.ebuild
Comment (juippis) [inline x11-libs/xapps/xapps-2.0.5.ebuild:82]:
```suggestion
		cp -r "${D}"/pygobject/* "${ED}/${PYTHON_GI_OVERRIDESDIR}/" || die
```

File: x11-libs/xapps/xapps-2.0.5.ebuild
Comment (juippis) [inline x11-libs/xapps/xapps-2.0.5.ebuild:83]:
```suggestion
		python_optimize "${ED}/${PYTHON_GI_OVERRIDESDIR}/"
```

File: x11-libs/xapps/xapps-2.0.5.ebuild
Comment (juippis) [inline x11-libs/xapps/xapps-2.0.5.ebuild:86]: As explained before, drop the `-f`. Although maybe for this one it's fine, but try without. We want to see if something doesn't go as planned.

File: x11-wm/muffin/muffin-4.8.0.ebuild
Comment (juippis) [inline x11-wm/muffin/muffin-4.8.0.ebuild:6]: I didn't see `eutils` being used in here. And it shouldnt't have any ebuild usage left anymore, probably a remnant?

File: x11-wm/muffin/muffin-4.8.0.ebuild
Comment (juippis) [inline x11-wm/muffin/muffin-4.8.0.ebuild:9]: Add Github page.

File: x11-wm/muffin/muffin-4.8.0.ebuild
Comment (juippis) [inline x11-wm/muffin/muffin-4.8.0.ebuild:44]: No newline needed.

File: x11-wm/muffin/muffin-4.8.0.ebuild
Comment (juippis) [inline x11-wm/muffin/muffin-4.8.0.ebuild:50]: No newline needed.

File: x11-wm/muffin/muffin-4.8.0.ebuild
Comment (juippis) [inline x11-wm/muffin/muffin-4.8.0.ebuild:64]: This is not really needed with EAPI-7 as it already handles the environment resets, but it makes really no difference. Just FYI.

**Review by juippis — COMMENTED** [review]
Hey, 

honestly I didn't see anything big wrong with this, apart from couple "real" issues. Most of what I've commented is just style nitpicking, since I decided there'd be too much for myself to fix after merging. So good job here! I can squash the commits after you if you don't want to rebase, it'd probably make re-reviewing a bit easier for me too. So if you wish, please do add a separate commit per-package fixing whatever there is, and I'll re-review and squash them into your original commits.

**Comment by vovan69** [comment]
I've tried these new 4.8 ebuilds. Compilation was successful, but cinnamon was crashing at startup and switching to the fallback mode. I've tried "test" user with clean home directory as well and spent some time looking into log files to understand the reason, but haven't been able to identify...

File: x11-wm/muffin/muffin-4.8.0.ebuild
Comment (SparkyBluefang) [reply x11-wm/muffin/muffin-4.8.0.ebuild]: Is this documented somewhere? I didn't see any references in the EAPI documentation, eclasses, or in portage's installed files. I would have expected `xdg_environment_reset` to have been banned in EAPI-7 if this was the case.

File: gnome-extra/cinnamon-translations/cinnamon-translations-4.8.3.ebuild
Comment (SparkyBluefang) [reply gnome-extra/cinnamon-translations/cinnamon-translations-4.8.3.ebuild:8]: That policy seems a little unclear to me. It seems like its targeted at packages that take a long time to rebuild (i.e. rust, libreoffice, gcc, boost), and small files (which I take to be both small in individual size and in total count).

It seems like localization packages (i.e. libreoffice-l10n, cinnamon-transaltions) wouldn't exactly fit that definition, as they are quick to build and would install hundreds of files that take 10s of MBs.

File: gnome-extra/cinnamon-control-center/cinnamon-control-center-4.8.2.ebuild
Comment (SparkyBluefang) [reply gnome-extra/cinnamon-control-center/cinnamon-control-center-4.8.2.ebuild:31]: I there a codified style guide somewhere?

File: x11-wm/muffin/muffin-4.8.0.ebuild
Comment (juippis) [reply x11-wm/muffin/muffin-4.8.0.ebuild]: https://mgorny.pl/articles/the-ultimate-guide-to-eapi-7.html#environment-variable-filtering-env-unset

File: gnome-extra/cinnamon-control-center/cinnamon-control-center-4.8.2.ebuild
Comment (juippis) [reply gnome-extra/cinnamon-control-center/cinnamon-control-center-4.8.2.ebuild:31]: https://wiki.gentoo.org/wiki/Project:Proxy_Maintainers/User_Guide/Style_Guide#Sorting_dependencies

File: gnome-extra/cinnamon-session/cinnamon-session-4.8.0.ebuild
Comment (juippis) [inline gnome-extra/cinnamon-session/cinnamon-session-4.8.0.ebuild:54]: I'm not sure what this USE flag does, 
```
/usr/share
/usr/share/cinnamon-session
/usr/share/cinnamon-session/csm-inhibit-dialog.glade
/usr/share/cinnamon-session/hardware-compatibility
/usr/share/doc
/usr/share/doc/cinnamon-session
/usr/share/doc/cinnamon-session-4.8.0
/usr/share/doc/cinnamon-session-4.8.0/AUTHORS.bz2
/usr/share/doc/cinnamon-session-4.8.0/README.bz2
/usr/share/doc/cinnamon-session-4.8.0/README.md
/usr/share/doc/cinnamon-session/dbus
/usr/share/doc/cinnamon-session/dbus/cinnamon-session.html
```
Maybe the files don't get installed?

Comment (juippis) [inline gnome-extra/cinnamon-session/cinnamon-session-4.8.0.ebuild:54]:
```
 * Messages for package gnome-extra/cinnamon-session-4.8.0:

 * The ebuild is installing to one or more unexpected paths:
 * 
 *   /usr/share/doc/cinnamon-session
 * 
 * Please fix the ebuild to use correct FHS/Gentoo policy paths.
```
Seems to be an empty dir though.

**Review by juippis — APPROVED** [review]

**Pull request closed without merging**
